plugins/nanospell/plugin.js

```diff
@@ -18,769 +18,774 @@
-                        return false;
-                    }
-                },
-                removeFormatFilterTemplate = function (element) {
-                    var result = true;
-
-                    if (element.hasClass('nanospell-typo')) {
-                        result = false;
-                    }
-
-                    return result;
-                };
-
-            if (pathFilters) {
-                pathFilters.push(pathFilter);
-            }
-
-            if (dataFilter) {
-                var dataFilterRules = {
-                    elements: {
-                        span: function (element) {
-
-                            var scaytState = element.hasClass('nanospell-typo');
-
-                            if (scaytState) {
-                                delete element.name;
-                            }
-
-                            return element;
-                        }
-                    }
-                };
-
-                dataFilter.addRules(dataFilterRules);
-            }
-
-            if (htmlFilter) {
-                var htmlFilterRules = {
-                    elements: {
-                        span: function (element) {
-
-                            var scaytState = element.hasClass('nanospell-typo');
-
-                            if (scaytState) {
-                                delete element.name;
-                            }
-
-                            return element;
-                        }
-                    }
-                };
-
-                htmlFilter.addRules(htmlFilterRules);
-            }
-
-            if (removeFormatFilter) {
-                removeFormatFilter.call(editor, removeFormatFilterTemplate);
-            }
-        }
-
-    });
+	'use strict';
+
+	var maxRequest = 200;
+	var editorHasFocus = false;
+	var spell_delay = 250;
+	var spell_fast_after_spacebar = true;
+	var settings_path;
+	var state = false;
+	var lang = "en";
+	var locale = {
+		ignore: "Ignore",
+		learn: "Add To Personal Dictionary",
+		nosuggestions: "( No Spelling Suggestions )"
+	};
+	var spellcache = [];
+	var suggestionscache = [];
+	var ignorecache = [];
+
+	function cleanQuotes(word) {
+		return word.replace(/[\u2018\u2019]/g, "'");
+	};
+
+	// wordwalker definition
+
+	function WordWalker(range) {
+		var isNotBookmark = CKEDITOR.dom.walker.bookmark(false, true);
+
+		var startNode = range.startContainer;
+		var endNode = range.endContainer;
+
+		console.assert(startNode.equals(endNode));
+
+		function evaluator(node) {
+			var path = new CKEDITOR.dom.elementPath( node, startNode );
+
+			return node.type == CKEDITOR.NODE_TEXT && // it is a text node
+				node.getLength() > 0 &&  // and it's not empty
+				( !node.isReadOnly() ) &&   // or read only
+				isNotBookmark(node) &&  // and isn't a fake bookmarking node
+				(path.block && path.block.equals(startNode)); // it's not nested in a deeper block from our root.
+		}
```

Comment:
This return (last part of it) is the only actual change, the rest of it is mostly moving things around.